swarm/network/stream/delivery.go

```diff
@@ -140,6 +140,17 @@ func (d *Delivery) handleRetrieveRequestMsg(sp *Peer, req *RetrieveRequestMsg) e
 	log.Trace("received request", "peer", sp.ID(), "hash", req.Key)
 	handleRetrieveRequestMsgCount.Inc(1)
 
+	// swap - record credit for 1 request
+	// note that only charge actual reqsearches
+	var err error
+	if sp.swap != nil {
```

Comment:
I am not sure we should introduce `swap` by checking if the value is `nil` everywhere it might be applicable. I think we should review our design/interfaces and abstract `swap` into an implementation which supports it.